### PR TITLE
Handle Exceptions in Apply() Function

### PR DESCRIPTION
**Pull request by danintel**
This prevents endless retry loops calling Apply() forever on exceptions.

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by danintel** [comment]
This fixes part of Issue #29 when the C++ TP Apply() function does not handle exceptions.
As a result, the Apply() function is called in a loop forever.
This fixes this issue.

It is still the case that an account has to be created  (with the "deposit" function) before a "transfer" is performed to it.

File: cxxprocessor/simplewallet-tp.cpp [hunk 1]
```diff
@@ -114,49 +114,56 @@ class SimpleWalletApplicator:  public sawtooth::TransactionApplicator {
...
+                std::cout << "Got beneficiary: " << beneficiary_pubkey << "\n";
+                this->doTransfer(customer_pubkey, amount, beneficiary_pubkey);
+            }
+            // Add your own action and a corresponding handler here
+            // Also add the actions in the client app as well
+            else {
+                std::string error = "Invalid action: '" + action + "'";
+                throw sawtooth::InvalidTransaction(error);
+            }
+        } catch(std::exception& e) {
```
Comment (askmish) [inline cxxprocessor/simplewallet-tp.cpp]: If we catch all the exceptions from this part of code, then exceptions related to valid errors like sawtooth::InvalidTransaction(error)  won't be printed onto debug console. Is that what we want to do here?

**Review by askmish — CHANGES_REQUESTED** [review]

File: cxxprocessor/simplewallet-tp.cpp (hunk 1 above)
Comment (danintel) [reply cxxprocessor/simplewallet-tp.cpp]: I added this code in `Apply()`:

`        } catch(sawtooth::InvalidTransaction& e) {`
`                throw;`
`        }`

**Review by askmish — APPROVED** [review]

**Comment by TheOnlyJoey** [comment]
Looks pretty solid, works locally for me.

**Pull request merged**